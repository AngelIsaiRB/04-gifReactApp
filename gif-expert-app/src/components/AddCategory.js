import React, { useState } from 'react'

export const AddCategory = () => {
    
    const [inputValue, setInputValue] = useState("hola mundo")
    const handleInputchange = (e)=>{
        setInputValue(e.target.value)
    }

    const handleSubmit =(e)=>{
        e.preventDefault();
        console.log("submi echoo")
    }


    return (
      
           <form onSubmit={handleSubmit}>
           <input
                type="text"
                value={inputValue}
                onChange={handleInputchange} 
           ></input>
           </form>

       
    )
}
