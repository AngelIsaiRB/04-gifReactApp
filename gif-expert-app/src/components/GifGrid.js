import React, { useEffect, useState } from 'react'
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({category}) => {
    
    const [images, setimages] = useState([]);

    useEffect(()=>{
        getGifs();
    },[])

    const getGifs =async ()=>{
        const url = "https://api.giphy.com/v1/gifs/search?q=halo&api_key=PK5OhhNxA3GgTlKxFjc2YAsyeouCAw5M&limit=10";
        const respuesta = await fetch(url);
        const {data}= await respuesta.json();
        const gifs = data.map( img =>{
            return {
                id:img.id,
                title:img.title,
                url:img.images?.downsized_medium.url
            }
        })
        console.log(gifs);
        setimages(gifs);
    };   

    return (
        <>
        <h3>{category}</h3>
        <div className="card-grid">
            
                {
                    images.map((img)=>{
                      return <GifGridItem 
                       key={img.id}
                      {...img}/>
                    })
                }
           
        </div>
        </>
    )
}
