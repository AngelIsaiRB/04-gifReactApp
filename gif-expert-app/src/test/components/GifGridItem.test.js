import React from 'react';
import  { shallow } from 'enzyme';
import { GifGridItem } from '../../components/GifGridItem';


describe('test en el componete de gidGridItem', () => {
    const title="un titulo";
    const url="https://unaimagen.jps";

    let wrapper = shallow(<GifGridItem title={title} url={url}/>);

    test('should de mostrtar el componente correctamente', () => {
            expect(wrapper).toMatchSnapshot();
    });

    test('should de tener un parrafo con el titulo', () => {
        const p = wrapper.find("p");
        expect(p.text().trim()).toBe(title);
    });

    test('should de tener la imagen iugal al url y alt de los props', () => {
        const img = wrapper.find("img");
        expect(img.prop("src")).toBe(url);
        expect(img.prop("alt")).toBe(title);
    });

    test('should de conter animacion ', () => {
        const div=wrapper.find("div");
        const constain = (div.prop("className")).includes("animate");
        expect(constain).toBe(true);
    });
    
    
    
    
    
})
