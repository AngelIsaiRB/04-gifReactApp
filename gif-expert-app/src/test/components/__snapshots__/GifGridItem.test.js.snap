// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test en el componete de gidGridItem should de mostrtar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__fadeIn "
>
  <img
    alt="un titulo"
    src="https://unaimagen.jps"
  />
  <p>
    un titulo
     
  </p>
</div>
`;
